webpack/utils/getEntrypoints.js

```diff
@@ -36,7 +36,7 @@ function findAssets(folder) {
     // Load old global module dependencies
     asset[fileName].unshift(path.join(basePath, "./src/lib/global_modules"))
     if (env.isDevelopment) {
-      asset[fileName].unshift("webpack-hot-middleware/client?reload=true")
```

Comment:
Why remove `reload`? If something can't hot reload (for example a change to a remote file in the codebase), this flag will trigger a page reload.

Reply:
During testing this cause unexplainable infinite loops instead of providing value. Is there anything specific that you need this functionality for. It's always preferable to leave features disabled until there is a need.

Reply:
> It's always preferable to leave features disabled until there is a need

This flag will automatically trigger a full-page reload if code changes in a distant part of a codebase. The need is -- page reloading when hot reloading isn't possible, its very useful so product devs don't have to manually reload and lose time. If its suddenly leading to some kind of infinite loop then a bug has been introduced in this PR related to our webpack config. 

(I can pull down branch and give it a look.)

Reply:
That didn't answer the question about if there is an actual need for it. Enabling it because it might be useful however we don't know why is a bad reason. If we know why, we can likely solve the problem a better way.

Reply:
After re-adding `reload=true` and testing out [on the /auctions page](https://github.com/artsy/force/blob/cac50409eaa7be4e4183926ad2f58f00f2d24daa/src/desktop/apps/auctions/client/index.coffee#L20), things work as expected. Can you provide an exact codepath to the infinite loop behavior described above? 

![reload](https://user-images.githubusercontent.com/236943/99581151-37ebd380-2995-11eb-9139-ee96d6257d20.gif)

Reply:
Once again, is this flag required for the change you demonstrated? If not let's move forward this may be added in the future if there become a real need for it.

Reply:
@damassi would you be opposed to me enabling hot reloading so that we don't need to force page reloads.

Reply:
Hot reloading works just great when we're inside of react code paths, v2 folder, etc. The reload is fallback feature for code outside of hot-reloadable paths, such as old apps. Webpack is intelligent enough to know when something isn't reloadable and will trigger that. 

Hot reloading example: 

![reload2](https://user-images.githubusercontent.com/236943/99596439-1a296900-29ab-11eb-9a81-ac75cc10998a.gif)

Reply:
I am well aware of how hot reloading works. A change is received, it walks up the module tree, looks for an acceptance handler, and if it reaches the entry point without being accepted it fails. Webpack only reloads code that we have explicitly defined as reloadable. The only reason hot reloading is not working in legacy code is that it is not enabled, CoffeeScript will hot reload fine.

Due to my familiarity with hot reloading in large applications, mainly all of the subtle ways it can introduce inconsistencies is why I've been asking you if there is a specific reason for the `reload` flag, e.g. internal state inconsistencies, sessions issues, templates patching incorrectly. From what I've seen so far, I believe that enabling hot reloading is a better, faster solution.

Additionally, hot reloading won't work for jade files even with `reload=true`. I'm okay with leaving the `reload` flag in place however if we are seeing frequent page reloads it means we should double-check that hot-reloading is properly configured. To clarify terminology when you say "fallback" you mean a local fallback for code that doesn't have hot reload enabled? By default, the hot middleware doesn't have a fallback.

p.s. The gifs are epic however without step-by-step instructions it's impossible to figure out what is actually going on.

Reply:
>  I'm okay with leaving the reload flag in place

Thanks, it's helpful for the team working in areas of the code that aren't new. We haven't had any reported problems with it. I don't think its worth it to figure out how to wire up for old code paths. This reload behavior works fine.

> To clarify terminology when you say "fallback" you mean a local fallback for code that doesn't have hot reload enabled? By default, the hot middleware doesn't have a fallback.

Meaning, if a file is edited that can't be hot reloaded it will fall back to a full page reload. Per [the docs](https://github.com/webpack-contrib/webpack-hot-middleware#client):
> Set to true to auto-reload the page when webpack gets stuck.

> The gifs are epic however without step-by-step instructions it's impossible to figure out what is actually going on

If you re-read my comments with screencaps i've left direct links to the files that have been edited so that you can play with it yourself. In [this comment](https://github.com/artsy/force/pull/6668#discussion_r526380228) theres a link to the file that I changed to trigger a full page reload; and in [this comment](https://github.com/artsy/force/pull/6668#discussion_r526468835) there's a link to the file showing hot reloading behavior. I'm not sure what more other than provide a screencap of whats happening and directly links to the source code.

Regarding reloads, here are all of [the options](https://github.com/webpack-contrib/webpack-hot-middleware#client) that we can pass the middleware. Sure, the fallback isn't enabled by default; that's why this prop was added.

Reply:
Sounds like we are good to merge!

> I'm not sure what more I can do other than provide a screencap of whats happening and direct links to the source code relevant to the screen cap.

Repro steps would have been more helpful, otherwise yes Zoom is the best option. Having never worked on consignment I have zero knowledge of how you accessed that page or how you constructed the scenario. The file provides me with a clue, though it leaves much detective work to reproduce what you are seeing.

1. Do this step.
2. Do this step.
3. ...

> Set to true to auto-reload the page when webpack gets stuck.

This is my favorite line in their documentation because it's so inexact. In our case, we aren't leveraging it for webpack errors, nor is it even stuck. If you read through the source the word "stuck" doesn't make sense.